php-lover/assignment7.py

```diff
@@ -0,0 +1,99 @@
+"""Calculate a given string equation
+
+python assignment7.py "(4+5)*4/6 - 1"
+
+"""
+import re
+import sys
+
+class Calculator:
+    def __init__(self):
+        self.OPS = {
+            '/': self.div,
+            '*': self.mul,
+            '-': self.sub,
+            '+': self.add
+        }
+
+
+    def calculate(self, equation: str) -> float:
+        """Calculate a string equation using PEMDAS rule"""
+        nq = []
+        oq = []
+
+        l = re.split(r'\s|([\/*+\-\(\)])', equation)
+        eq_arr = [i for i in l if i != '' and i is not None]
+
+        i = 0
+        while i < len(eq_arr):
+            # create the equation queue 
+            x = eq_arr[i]
+            if x == '(':
+                # process parentheses
+                start = i+1
```

Comment:
Python has officially published style guidelines and I believe they specify to have spaces around operators.

Reply:
resolved